api/v1alpha1/virtualbackend_types.go

```diff
@@ -0,0 +1,32 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+const (
+	// KindSecurityPolicy is the name of the SecurityPolicy kind.
+	KindVirtualBackend = "VirtualBackend"
+)
+
+type Body string
+
+type StatusCode uint32
+
+type ResponseHeader string
+
+// VirtualBackend defines the configuration for direct response.
+type VirtualBackend struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              VirtualBackendSpec `json:"spec"`
+}
+
+type VirtualBackendSpec struct {
```

Comment:
Add some CEL validations here, and also document what the defaults are if nothing is provided.

For consistency with other EG APIs, if the value is optional:
* It should be a pointer 
* The JSON tag should specify "omitempty"
* There should be some documentation around what is used if the value was not specified.

This is especially important for the `StatusCode` field - there might not be a body but the direct response should always include a `StatusCode` - it shouldn't be possible to leave this empty.

Reply:
@liorokman done.